Assets/Scripts/Client Scripts/UIManager.cs

```diff
@@ -202,34 +207,94 @@ public void UpdateCurrentPlayerColor(int player)
     }
     #endregion
 
+
+    private void Update()
+    {
+        if (Input.GetKeyDown(KeyCode.A))
+        {
+            ToggleRevealUI();
+        }
+    }
     #region Reveal UI
     public void InitializeRevealButtons()
     {  
         //this is a terrible way to do this. Arjun and Evan should talk about the best way to solve this.
-        goal1.GetComponentInParent<Button>().onClick.AddListener(() => { OnRevealClick(false);});
-        goal2.GetComponentInParent<Button>().onClick.AddListener(() => { OnRevealClick(true);});
-        goal1.text = GameManager.cursors[goal_1.id].username;
-        goal2.text = GameManager.cursors[goal_2.id].username;
+        revealedGoals[Client.instance.myId-1].transform.GetChild(1).GetComponent<Button>().onClick.AddListener(() => { OnRevealClick(false);});
+        revealedGoals[Client.instance.myId-1].transform.GetChild(2).GetComponent<Button>().onClick.AddListener(() => { OnRevealClick(true);});
+        revealedGoals[Client.instance.myId - 1].transform.GetChild(1).GetComponent<TMP_Text>().text = $"{goal_1.goalState} {GameManager.cursors[goal_1.id].username}";
+        revealedGoals[Client.instance.myId - 1].transform.GetChild(2).GetComponent<TMP_Text>().text = $"{goal_2.goalState} {GameManager.cursors[goal_2.id].username}";
     }
     
 
     public void OnRevealClick(bool right)
     {
-        currentGoals.text += GameManager.cursors[Client.instance.myId].username + ": ";
+        //currentGoals.text += GameManager.cursors[Client.instance.myId].username + ": ";
         if (right)
         {
-            currentGoals.text += goal2.text + "\n";
+            //currentGoals.text += goal2.text + "\n";
             ClientSend.RevealTarget(goal_1);
         }
         else
         {
-            currentGoals.text += goal1.text + "\n";
+            //currentGoals.text += goal1.text + "\n";
             ClientSend.RevealTarget(goal_2);
         }
         goal1.GetComponentInParent<Button>().gameObject.SetActive(false);
         goal2.GetComponentInParent<Button>().gameObject.SetActive(false);
         // Initialize Next UI
         // Talk to Evan if we should seperate the 2 UI Canvas
     }
+
+    public void NextGoalLine()
+    {
+        revealedGoals[currentPlayer].SetActive(true);
+        currentPlayer++;
+    }
+    public void WriteRevealedGoal(Goal goal)
+    {
```

Comment:
Just add note to be careful when adding to the revealedGoals prefab (named: prefabGoalEntryUnfinish)